### PR TITLE
Remove MINPVFIL

### PR DESCRIPTION
**Pull request by OPMUSER**
MINPVFIL is an OPM Flow specific keyword that is not used by the simulator,  and thus it should be deleted.

Previously, we moved the checks that only allow either the MINPV or MINPORV keywords to be used in the input deck to the JSON keyword files.  This was done to avoid confusion over which global limits are being applied.

Note that if the MINPVV keyword is also in the input deck then these limits are applied post processing the global limits defined by the MINPV/MINPORV keywords. This behavior is unchanged by this PR.

The PR is marked as WIP to give developers time to object for the removal of MINPVFIL keyword.

### CODE REVIEW

**Comment by alfbr** [comment]
Actually, MINPVFIL was used for the original MINPV implementation in Flow. The first implementation of MINPV did not only remove the cell with volume below the threshold, but it moved the top of the cell below, or the bottom of the cell above, to fill in the void. Later, MINPV was implemented according to the keyword definition, and then the original functionality was kept as the custom keyword MINPVFIL. That code path may be lost now, and I am not sure it is being used anymore. Hence, I do not object to removing it. Still, I do believe it was a better way of accomplishing what we use MINPV+PINCH for (although not as flexible as that combination with its options).

**Comment by bska** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
I also do not mind removing MINPVFIL. As @alfbr said above, it had some merits, but the cost of maintaining it separately from the MINPV and PINCH features was not worth it.

**Comment by OPMUSER** [comment]
Hopefully this can be merged.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
I'm too slow. logs already gone.

**Comment by blattms** [comment]
jenkins build this please

File: tests/parser/EclipseGridTests.cpp [hunk 1]
```diff
@@ -847,7 +847,7 @@ BOOST_AUTO_TEST_CASE(ConstructorMINPV) {
     auto deck3 = createMinpvCPDeck();
 
     Opm::EclipseGrid grid1(deck1);
-    BOOST_CHECK_THROW(Opm::EclipseGrid grid2(deck2), std::invalid_argument);
+    BOOST_CHECK_THROW(Opm::EclipseGrid grid2(deck2), OpmInputError);
```
Comment (blattms) [inline tests/parser/EclipseGridTests.cpp:850]: I think this needs to be Opm::OpmInputError for fixing the error:
```
/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/tests/parser/EclipseGridTests.cpp: In member function 'void ConstructorMINPV::test_method()':
/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/tests/parser/EclipseGridTests.cpp:849:54: error: 'OpmInputError' does not name a type
  849 |     BOOST_CHECK_THROW(Opm::EclipseGrid grid2(deck2), OpmInputError);
      |                                                      ^~~~~~~~~~~~~
/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/tests/parser/EclipseGridTests.cpp:849:5: error: expected ')' before 'const'
  849 |     BOOST_CHECK_THROW(Opm::EclipseGrid grid2(deck2), OpmInputError);
      |     ^~~~~~~~~~~~~~~~~
/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/tests/parser/EclipseGridTests.cpp:849:5: error: expected '{' before 'const'
/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/tests/parser/EclipseGridTests.cpp:849:5: error: expected initializer before ')' token
  849 |     BOOST_CHECK_THROW(Opm::EclipseGrid grid2(deck2), OpmInputError);
      |     ^~~~~~~~~~~~~~~~~
```

File: tests/parser/EclipseGridTests.cpp [hunk 2]
```diff
@@ -255,7 +255,7 @@ static Opm::Deck createMinpvDefaultCPDeck() {
         "  1000*1 / \n"
         "MINPV \n"
         "  / \n"
-        "MINPVFIL \n"
+        "MINPORV \n"
```
Comment (blattms) [inline tests/parser/EclipseGridTests.cpp:258]: This will throw in line 846, because the combination of MINPV and MINPORV is not allowed.

**Comment by blattms** [comment]
Closing, because fixed version #3461 was merged.

**Pull request closed without merging**